packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -198,35 +200,78 @@ export function PanelChrome({
 
   // Handle drag & selection events
   // Mainly the tricky bit of differentiating between dragging and selecting
+  // mouseEnter and mouseLeave are necessary to avoid triggering dragging by mistake on various scenarios (i.e., text selection on screen)
+  // The mechanism is pretty straightforward:
+  //   - when clicking the pointer, mark the click
+  //   - if we leave/reenter the area, we don't trigger neither the select nor the drag
+  //   - if we move > 10 units, we trigger the drag
+  //   - if we move <= 10 units, we trigger the select
+
+  const resetPointerStatus = () => {
+    pointerStatus.current = { pointerDown: false, dragStarted: false, initialPosition: { x: 0, y: 0 } };
+  };
 
-  const onPointerUp = (evt: React.PointerEvent) => {
-    evt.stopPropagation();
+  const onHeaderEnter = React.useCallback(() => {
+    resetPointerStatus();
 
-    const distance = Math.hypot(
-      pointerDownPos.current.screenX - evt.screenX,
-      pointerDownPos.current.screenY - evt.screenY
-    );
+    if (isSelectable && !selectableHighlight) {
+      setSelectableHighlight(true);
+    }
+  }, [isSelectable, selectableHighlight]);
 
-    pointerDownPos.current = { screenX: 0, screenY: 0 };
+  const onHeaderLeave = React.useCallback(() => {
+    resetPointerStatus();
 
-    // If we are dragging some distance or clicking on elements that should cancel dragging (panel menu, etc)
-    if (
-      distance > 10 ||
-      (dragClassCancel && evt.target instanceof HTMLElement && evt.target.closest(`.${dragClassCancel}`))
-    ) {
-      return;
+    if (isSelectable && selectableHighlight) {
+      setSelectableHighlight(false);
     }
+  }, [isSelectable, selectableHighlight]);
 
-    onSelect?.(evt);
-  };
+  const onPointerUp = React.useCallback(
+    (evt: React.PointerEvent) => {
+      const dragStarted = pointerStatus.current.dragStarted;
 
-  const onPointerDown = (evt: React.PointerEvent) => {
-    evt.stopPropagation();
+      resetPointerStatus();
 
-    pointerDownPos.current = { screenX: evt.screenX, screenY: evt.screenY };
+      if (
+        dragStarted ||
+        (dragClassCancel && evt.target instanceof Element && evt.target.closest(`.${dragClassCancel}`))
+      ) {
+        return;
+      }
```

Comment:
I feel like we should select panels we are dragging

Reply:
100%